lib/API/JsonApi/v1/Filters/Interpreters/FilterInterpreter.php

```diff
@@ -0,0 +1,332 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with
+ * this program; if not, see http://www.gnu.org/licenses or write to the Free
+ * Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA.
+ *
+ * You can contact SugarCRM, Inc. headquarters at 10050 North Wolfe Road,
+ * SW2-130, Cupertino, CA 95014, USA. or at email address contact@sugarcrm.com.
+ *
+ * The interactive user interfaces in modified source and object code versions
+ * of this program must display Appropriate Legal Notices, as required under
+ * Section 5 of the GNU Affero General Public License version 3.
+ *
+ * In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+ * these Appropriate Legal Notices must retain the display of the "Powered by
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+
+namespace SuiteCRM\API\JsonApi\v1\Filters\Interpreters;
+
+use SuiteCRM\API\JsonApi\v1\Filters\Interfaces\OperatorInterface;
+use SuiteCRM\API\JsonApi\v1\Filters\Interpreters\ByIdFilters\ByIdFilter;
+use SuiteCRM\API\JsonApi\v1\Filters\Operators\FieldOperator;
+use SuiteCRM\API\JsonApi\v1\Filters\Operators\Operator;
+use SuiteCRM\API\JsonApi\v1\Filters\Validators\FieldValidator;
+use SuiteCRM\API\v8\Exception\BadRequest;
+use SuiteCRM\Exception\Exception;
+
+use Psr\Container\ContainerInterface;
+
+class FilterInterpreter
+{
+    /**
+     * @var ContainerInterface $containers
+     */
+    private $containers;
+    private static $fieldOperator;
+    private static $filterOperators;
+    private static $filterFieldOperators;
+    private static $filterSpecialOperators;
+
+    /**
+     * FilterInterpreter constructor.
+     * @param ContainerInterface $containers
+     * @throws \Psr\Container\ContainerExceptionInterface
+     */
+    public function __construct(ContainerInterface $containers)
+    {
+        $this->containers = $containers;
+
+        if (self::$fieldOperator === null) {
+            self::$fieldOperator = new FieldOperator($containers);
+        }
+
+        if (self::$filterOperators === null) {
+            self::$filterOperators = $containers->get('FilterOperators');
+        }
+
+        if (self::$filterFieldOperators === null) {
+            self::$filterFieldOperators = $containers->get('FilterFieldOperators');
+        }
+
+        if (self::$filterSpecialOperators === null) {
+            self::$filterSpecialOperators = $containers->get('FilterSpecialOperators');
+        }
+    }
+
+    /**
+     * @param $filterStructure
```

Comment:
class or type?

Reply:
array ... we may turn this into a class later ... I will update the php docs